### ISSUE TITLE
`UP035` doesn't detect `typing_extensions.override` when targeting python 3.12

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ruff#9443]
```py
from typing_extensions import override  # no error
```
https://play.ruff.rs/f12262a2-a134-4035-8901-6e1affb88555

`override` was added to the `typing` module in 3.12, so ruff should complain here
